### PR TITLE
{cdn} Upgrade cdn api version to 2024-09-01

### PR DESCRIPTION
**Pull request by Ptnan7**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by vidai-msft** [comment]
@Ptnan7 Our CI detected this PR has breaking changes. Can you please confirm? Please also fix the examples issues as described in the CI.

File: src/Cdn/Cdn.Autorest/README.md [hunk 1]
```diff
@@ -51,9 +51,12 @@ identity-correction-for-post: true
 resourcegroup-append: true
 nested-object-to-string: true
 
+disable-transform-identity-type: true
+flatten-userassignedidentity: false
+
 # For new modules, please avoid setting 3.x using the use-extension method and instead, use 4.x as the default option
 use-extension:
-  "@autorest/powershell": "3.x"
+  "@autorest/powershell": "4.x"
```
Comment (vidai-msft) [inline src/Cdn/Cdn.Autorest/README.md:59]: 4.x is the by default value. If you plan to migrate to 4, then can remove the `use-extension` attribute.

File: src/Cdn/Cdn.Autorest/README.md [hunk 2]
```diff
@@ -304,4 +484,24 @@ directive:
     where: $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/endpoints/{endpointName}"].delete
     transform: >-
       $["x-ms-long-running-operation-options"] = {"final-state-via": "azure-async-operation"}
+
+  # announce upcoming MI-related breaking changes
+  - where:
+      parameter-name: IdentityType
+    set:
+      breaking-change:
```
Comment (vidai-msft) [inline src/Cdn/Cdn.Autorest/README.md:492]: Are you planning to roll out breaking changes in this Ignite or just the pre-announcement? I see these information mismatches. We will release Az 13 during Ignite. However, the pre-announcement should be released 2 or 3 sprints before the major release. It is not allowed to roll out breaking changes directly without announcement.

File: src/Cdn/Cdn.Autorest/README.md (hunk 1 above)
Comment (Ptnan7) [reply src/Cdn/Cdn.Autorest/README.md:59]: Removed.

**Pull request closed without merging**